packages/scene-composer/src/store/slices/SceneDocumentSlice.ts

```diff
@@ -384,30 +383,26 @@ export const createSceneDocumentSlice = (set: SetState<RootState>, get: GetState
           draft.document.properties = {};
         }
         draft.document.properties[property] = value;
-        draft.lastOperation = 'setSceneProperty';
-      });
-    },
 
-    clearTemplatizedDataBindings: () => {
-      set((draft) => {
-        Object.values(draft.document.nodeMap)
-          .flatMap((node) => node.components)
-          .filter(
-            (component) =>
-              component.type === KnownComponentType.Tag || component.type === KnownComponentType.ModelShader,
-          )
-          .forEach((component) => {
-            const dataBindingContext = (component as IDataBoundSceneComponentInternal)?.valueDataBinding;
-            const hasDataBindingTemplate =
-              Object.values((dataBindingContext?.dataBindingContext as any) ?? {}).filter((value) =>
-                isDataBindingTemplate(value as string),
-              ).length > 0;
-            if (hasDataBindingTemplate) {
-              dataBindingContext!.dataBindingContext = {};
-            }
-          });
-
-        draft.lastOperation = 'clearTemplatizedDataBindings';
+        if (property === KnownSceneProperty.DataBindingConfig) {
+          Object.values(draft.document.nodeMap)
+            .flatMap((node) => node.components)
+            .filter(
+              (component) =>
+                component.type === KnownComponentType.Tag || component.type === KnownComponentType.ModelShader,
+            )
+            .forEach((component) => {
+              const dataBindingContext = (component as IDataBoundSceneComponentInternal)?.valueDataBinding;
+              const hasDataBindingTemplate =
+                Object.values(dataBindingContext?.dataBindingContext ?? {}).filter((value) =>
+                  isDataBindingTemplate(value as string),
+                ).length > 0;
+              if (hasDataBindingTemplate) {
+                dataBindingContext!.dataBindingContext = undefined;
+              }
+            });
+        }
+        draft.lastOperation = 'setSceneProperty';
```

Comment:
This section appears to be pre-existing, yet we should consider refactoring or commenting in a future change. There's a side-effect which isn't obvious at first glance until you notice the return of `Object.values` is not being assigned, which makes me feel like there could be a potential bug in the future here.

Reply:
you mean the `Object.values(draft.document.nodeMap)` part? yeah.. but things are working fine right now and change it could potentially cause a lot more unexpected rerenderings. I'll keep it now until we find any bug caused by it

Reply:
> The nested check for "dataBindingContext" that has a databinding template should use find instead of filter,

will do